src/plonkish/compiler/mod.rs

```diff
@@ -556,3 +556,99 @@ fn add_halo2_columns<F, TraceArgs>(unit: &mut CompilationUnit<F>, ast: &astCircu
     unit.columns.extend(halo2_advice_columns);
     unit.columns.extend(halo2_fixed_columns);
 }
+
+#[cfg(test)]
+mod test {
+    use halo2_proofs::plonk::Any;
+    use halo2curves::bn256::Fr;
+
+    use super::{cell_manager::SingleRowCellManager, step_selector::SimpleStepSelectorBuilder, *};
+
+    #[test]
+    fn test_compiler_config_initialization() {
+        let cell_manager = SingleRowCellManager::default();
+        let step_selector_builder = SimpleStepSelectorBuilder::default();
+
+        let config = config(cell_manager.clone(), step_selector_builder.clone());
+
+        assert_eq!(
+            format!("{:#?}", config.cell_manager),
+            format!("{:#?}", cell_manager)
+        );
+        assert_eq!(
+            format!("{:#?}", config.step_selector_builder),
+            format!("{:#?}", step_selector_builder)
+        );
+    }
+
+    #[test]
+    fn test_compile() {
+        let cell_manager = SingleRowCellManager::default();
+        let step_selector_builder = SimpleStepSelectorBuilder::default();
+        let config = config(cell_manager, step_selector_builder);
+
+        let mock_ast_circuit = astCircuit::<Fr, Any>::default();
+
+        let (circuit, assignment_generator) = compile(config, &mock_ast_circuit);
+
+        assert_eq!(circuit.columns.len(), 1);
+        assert_eq!(circuit.exposed.len(), 0);
+        assert_eq!(circuit.polys.len(), 0);
+        assert_eq!(circuit.lookups.len(), 0);
+        assert_eq!(circuit.fixed_assignments.len(), 1);
+        assert_eq!(circuit.ast_id, mock_ast_circuit.id);
+
+        assert!(assignment_generator.is_none());
+    }
+
+    #[test]
+    fn test_compile_phase1() {
+        let cell_manager = SingleRowCellManager::default();
+        let step_selector_builder = SimpleStepSelectorBuilder::default();
+        let config = config(cell_manager, step_selector_builder);
+
+        let mock_ast_circuit = astCircuit::<Fr, Any>::default();
+
+        let (unit, assignment_generator) = compile_phase1(config, &mock_ast_circuit);
+
+        assert_eq!(unit.columns.len(), 1);
+        assert_eq!(unit.exposed.len(), 0);
+        assert_eq!(unit.polys.len(), 0);
+        assert_eq!(unit.lookups.len(), 0);
+        assert_eq!(unit.fixed_assignments.len(), 0);
+        assert_eq!(unit.ast_id, mock_ast_circuit.id);
+
+        assert!(assignment_generator.is_none());
+    }
+
+    #[test]
+    fn test_compile_phase2() {
+        let mock_ast_circuit = astCircuit::<Fr, Any>::default();
+
+        let mut unit = CompilationUnit::from(&mock_ast_circuit);
```

Comment:
I don't think it makes sense to compile phase2 wihtout doing phase1 before. Perhaps we should have an error if this happens.

Reply:
Agreed, so should we create an issue to fix this by tracking the phase of the compilation and ensuring that is always sequential? It should be quick but since it goes out of the unit tests scope maybe it's better to separate concerns

Reply:
Yes we could add a flag to compilation unit for this

Reply:
Created a new issue for this: https://github.com/privacy-scaling-explorations/chiquito/issues/192
Can I have that assigned and can we merge this PR? @leolara

Reply:
@rutefig Still, the result we need to test is after compile_phase1, then compile_phase2, please dont leave the test that does not execture phase1 before phase2